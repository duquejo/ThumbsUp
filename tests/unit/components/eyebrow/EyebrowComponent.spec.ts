import { describe, it, expect, beforeEach, vi } from 'vitest';
import { VueWrapper, shallowMount } from '@vue/test-utils';
import EyebrowComponentVue from '@/components/eyebrow/EyebrowComponent.vue';
import { formatDistanceToNow } from 'date-fns';

vi.mock('date-fns', () => ({
  formatDistanceToNow: vi.fn(),
}));

const mockedFormatDistanceToNow = vi.mocked(formatDistanceToNow);

describe('+ EyebrowComponent unit test', () => {

  const properties = {
    category: 'entertainment',
    lastUpdated: '2020-03-10T23:08:57.892Z',
  };

  let wrapper: VueWrapper;

  beforeEach(() => {
    wrapper = shallowMount(EyebrowComponentVue, {props: {  ...properties}});
  });
  
  it('should render properly', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should map the given props', () => {
    const time = wrapper.find('.eyebrow time');
    const category = wrapper.find('.eyebrow span');

    mockedFormatDistanceToNow.mockReturnValueOnce('almost 4 years ago');

    expect(time.text()).toEqual(expect.any(String));
    expect(category.text()).toBe(properties.category);

    expect(mockedFormatDistanceToNow).toHaveBeenCalled();
    expect(mockedFormatDistanceToNow).toHaveBeenCalledWith(properties.lastUpdated, expect.any(Object));
  });
});
