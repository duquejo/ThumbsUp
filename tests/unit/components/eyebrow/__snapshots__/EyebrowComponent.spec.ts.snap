// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`+ EyebrowComponent unit test > should render properly 1`] = `"<span class="eyebrow"><time></time> in <span>entertainment</span></span>"`;
